packages/block-library/src/utils/use-template-context.js

```diff
@@ -0,0 +1,325 @@
+/**
+ * WordPress dependencies
+ */
+import { store as coreStore } from '@wordpress/core-data';
+import { useSelect } from '@wordpress/data';
+
+const DEFAULT_CONTEXT = {
+	taxonomy: null,
+	termSlug: null,
+	isAuthor: false,
+	authorSlug: null,
+};
+
+/**
+ * Creates a template context object by merging provided context with default context.
+ *
+ * @param {Object} providedContext Provided context object to merge with default context.
+ * @return {Object} Complete context object with all required fields.
+ */
+function buildTemplateContext( providedContext = {} ) {
```

Comment:
Since all `DEFAULT_CONTEXT` values are falsy, why do we need this helper? Each consumer could just check for the values they need, no?

Reply:
Yes true, I'm not sure this helper is necessary. Removed in https://github.com/WordPress/gutenberg/pull/72549/commits/a990d963982846e00db96c3f28915489dd3c66b1.